src/python/move_emojis/emojis_to_const.py

```diff
@@ -0,0 +1,305 @@
+# cspell:words unidecode
+# cspell:words alnum
+
+# Helper file for converting all of the data in the Unicode Emoji data files
+# to json files.
+
+import json
+import re
+import string
+import sys
+
+from unidecode import unidecode
+
+import code_point
+
+BASE_EMOJI_DATA_PATH = "data/emoji-test.txt"
+VIABLE_EMOJI_JSON_PATH = "data/viable_emojis.json"
+ZWJ_EMOJIS_DATA_PATH = "data/emoji-zwj-sequences.txt"
+
+# In bytes.
+MAX_SYMBOL_SIZE = 10
+
+DEBUG = False
+
+
+def debug_print(*args, **kwargs):
+    if DEBUG:
+        print("[DEBUG]:", *args, **kwargs)
+
+
+def format_code_points_for_json(code_points: list) -> dict:
+    num_bytes = sum([len(code_point.to_bytes(cp)) for cp in code_points])
+    as_hex = [code_point.to_bytes(cp).hex() for cp in code_points]
+    return {
+        "num_bytes": num_bytes,
+        "as_unicode": code_points,
+        "as_hex": as_hex,
+    }
+
+
+def sanitized_lines(filename: str) -> list:
+    lines = []
+    with open(filename) as in_file:
+        for line in in_file.readlines():
+            line = line.strip()
+            if len(line) < 1 or line.startswith("#"):
+                continue
+
+            # Replace ‘ and ’ by using unicode code_points
+            # so the linter doesn't complain.
+            line = line.replace(code_point.to_emoji("2018"), "'")
+            line = line.replace(code_point.to_emoji("2019"), "'")
+            line = line.replace("“", '"')
+            line = line.replace("”", '"')
+            lines.append(line)
+    return lines
+
+
+def get_base_emojis(filename: str) -> dict:
+    lines = sanitized_lines(filename)
+
+    emoji_pattern = r"(.+)"
+    version_pattern = r"(E[0-9.]+)"
+    name_pattern = r"(.+)"
+    comment_pattern = rf"{emoji_pattern}\s+{version_pattern}\s+{name_pattern}"
+    comment_regex = re.compile(comment_pattern)
+
+    d: dict = {}
+    for line in lines:
+        semicolon_split = [s.strip() for s in line.split(";")]
+        # Order matters for code_points.
+        code_points = semicolon_split[0].split(" ")
+        comment_split = [s.strip() for s in semicolon_split[1].split("#", 1)]
+
+        qualification = comment_split[0]
+
+        comment = comment_split[1]
+        matches = comment_regex.match(comment)
+        if matches is None or len(matches.groups()) != 3:
+            print(f"ERROR: malformed line: {line}")
+            sys.exit(1)
+            continue
```

Comment:
Code is unreachable per `pylance`

Maybe we get a CI task for this on the v5 repo? Has come up a couple of times